### DISCUSSION TITLE
What about requirements to run for DIY people?

### DISCUSSION THREAD

**Discussion by msdobrescu** [LinSoftWin/Photoshop-CC2022-Linux#41] [Ideas]
Hi, I've seen here discussions that something does not work for some reason, sometimes unknown.
Let's publish the list of requirements for manual installation, like what env vars, fonts, runtimes (Visual Studio, .net etc.), that would be instaled with Winetricks and so on - stuff done by the script.
May help the users to know what to is needed install.

**Comment by MiMillieuh** [comment]
Well if you need everything to run it you have to download :

- wine
- my custom version of photoshop : https://lulucloud.mywire.org/FileHosting/GithubProjects/AdobePhotoshop2021.tar.xz
- vkd3d
- winetricks (fontsmooth=rgb gdiplus msxml3 msxml6 atmlib corefonts dxvk win10)
- vcredist 2010 2012 2013 2019

Launcher and desktop entry are optional. it's remanded to do a separated wine prefix
Since I’ve modified some file in Photoshop you can't really use your own... It will or crash or have a lot of issues

And you should be good to go.

Please note that any use of my project should be credited if you use it in your own.

**Comment by msdobrescu** [comment]
I know, I have my own setup since long time, All I need is https://lulucloud.mywire.org/FileHosting/GithubProjects/AdobePhotoshop2021.tar.xz.
But the only problem to run PS is to pass the authentication, because that needs a browser, which is not supported in Wine. I have the confirmation that any browser should work, Chrome, Firefox or Edge, since Windows 11 dropped IE.
So to be legit, should be able to run one of the browsers and use a vanilla PS. I've tried a lot to install Edge or Chrome or Firefox, but all fail in a way or another.

**Reply by MiMillieuh** [reply]
Sadly you can't, even if windows 11 dropped IE some dlls still in windows to ensure compatibility and adobe hasn't updated to something else even with 2022 suite. so yeah it's not possible to download your own version from the Creative Cloud like you would do on Windows.

I'm a student and my school pays my Adobe license but it's not possible to use this license since the login is broken and CC create a lot of issue in Photoshop on Linux.

**Reply by msdobrescu** [reply]
It works with Chrome or Edge or Firefox now. Tested in Windows 11. I pay a subscription for PS (10 Eur without taxes/ month). Sadly, can't use it as I have ditched Windows.

**Reply by MiMillieuh** [reply]
What works is login redirection... but the adobe application relies on IE not the login you can use anything you want for this and it's not dependent from the app itself but the app (Creative cloud) is based on IE and still not able to run due to this

**Reply by msdobrescu** [reply]
I have installed it under a W11 test machine without IE. It works. Tried CC, installed PS through CC.

**Reply by MiMillieuh** [reply]
I know but **IE DLLS still are in Windows 11 even if there isn't IE anymore** This is not because you don't see a software that it isn't here or there isn't part of it in...
if IE was completely deleted from windows 11, A lot of programs would have stopped working... including adobe

**Reply by msdobrescu** [reply]
Last time I've tried was a previous version of PS, 2020 or 2021, that worked by installing it under Windows 10, copied the installation dirs and registries and was working perfectly until the login expired. It did not need the IE/system stuff for anything else.

**Reply by MiMillieuh** [reply]
> Last time I've tried was a previous version of PS, 2020 or 2021, that worked by installing it under Windows 10, copied the installation dirs and registries and was working perfectly until the login expired. It did not need the IE/system stuff for anything else.

It works but in my case it's pointless because we use the same license at school and so login expire everyday. 

Secondly, I've made some optimization in this custom package. 

And thirdly,  if cc tries to open in your prefix,  you can just reinstalle the whole thing. 

So yeah I've opted for something stable and not annoying each time I want to use photoshop. I don't want to have always a windows vm on my computer.

**Reply by msdobrescu** [reply]
On one hand, I dislike the solution because makes me dependent on Windows. On the other hand, the current solution is not clear to me, hacking is not legal, you may be prosecuted, also if you know yourself to do it is one thing, if taken from some other source is likely some malware is there too.
I'd love to see a correct solution from Wine by installing PS and running it in its therms. To bad no browser is possible to install in Wine along with the required libs to run CC/PS.

**Reply by MiMillieuh** [reply]
> On one hand, I dislike the solution because makes me dependent on Windows. On the other hand, the current solution is not clear to me, hacking is not legal, you may be prosecuted, also if you know yourself to do it is one thing, if taken from some other source is likely some malware is there too.
> I'd love to see a correct solution from Wine by installing PS and running it in its therms. To bad no browser is possible to install in Wine along with the required libs to run CC/PS.

Well it's not like if I don't own a license... I literally also pay for it...

But if you want to try get the login to work go ahead but it's time loss with the current state of wine...

Some users on github has CC 2018 also without login and they never had any problems with adobe... I guess they don't care linux isn't their marketshare since they don't really want to put any effort in it.

**Reply by msdobrescu** [reply]
I agree. Do you know what are the exact components it needs/uses, in therms of dlls or other binaries?

**Reply by MiMillieuh** [reply]
You should have all the info with what I said in the first message. 
However be aware that rendering html in Photoshop is really unstable and will cause a crash

**Reply by msdobrescu** [reply]
My question is about the exact components needed to make it work without any hacks, like the exact IE or Edge parts used by CC/PS to login and further proceed to the actual app.

**Reply by albakhtari** [reply]
Look man, just use a Windows VM, it'll save you loads of time and effort if you don't want to use a **cracked*** version of this software.

*not hacked, hacking means gaining unauthorised access to a computer system, while cracking is changing certain parts of the program's code for certain benefits

**Reply by msdobrescu** [reply]
The idea is, if some knows the components in question, we could also help Wine team to provide them or an alternative.

**Reply by albakhtari** [reply]
Just look inside the installer script.
Also, @MiMillieuh already posted the requirements for **their modified version of Ps**, and if we knew what the official version of Ps needed, then we wouldn't have modified Ps in the first place.
We know that it needs something to do with the new version of Internet Explorer but we don't know what else exactly.
